test/main.ml

```diff
@@ -12,14 +12,30 @@ let pp_ok () = Format.fprintf fmt "%a !@." pp_green "OK"
 
 let pp_error msg = Format.fprintf fmt "%a: %s !@." pp_red "FAILED" msg
 
-let test_file ~optimize f =
+let test_file mode filename =
   Format.fprintf fmt "testing %s: `%a`... "
-    (if optimize then "optimized file" else "file          ")
-    Fpath.pp f;
+    ( match mode with
+    | `Optimize -> "optimized file"
+    | `Formatted -> "formatted file"
+    | `Normal -> "file          " )
```

Comment:
Could you define the following instead:

```ocaml
type test_mode =
  | Normal
  | Fmt
  | Opt
```

and avoid the polymorphic variant ?